chacha20_poly1305/src/chacha20.rs

```diff
@@ -0,0 +1,419 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! The ChaCha20 stream cipher from RFC8439.
+
+/// The first four words (32-bit) of the ChaCha stream cipher state are constants.
+const WORD_1: u32 = 0x61707865;
+const WORD_2: u32 = 0x3320646e;
+const WORD_3: u32 = 0x79622d32;
+const WORD_4: u32 = 0x6b206574;
+
+/// The quarter round indexes of a ChaCha block function.
+const CHACHA_ROUND_INDICIES: [(usize, usize, usize, usize); 8] = [
+    // The first 4 rounds are columns of the 4x4 matrix.
+    (0, 4, 8, 12),
+    (1, 5, 9, 13),
+    (2, 6, 10, 14),
+    (3, 7, 11, 15),
+    // The next 4 rounds are diagonals of the 4x4 matrix.
+    (0, 5, 10, 15),
+    (1, 6, 11, 12),
+    (2, 7, 8, 13),
+    (3, 4, 9, 14),
+];
+
+/// The cipher's block size is 64 bytes.
+const CHACHA_BLOCKSIZE: usize = 64;
+
+/// A 256-bit secret session key shared by the parties communicating.
+#[derive(Clone, Copy, Debug)]
```

Comment:
If this is secret it shouldn't *derive* `Debug` but impl it using a placeholder (or hash but that's not required now).

Reply:
Makes sense, I'll drop all the `Debug`s since it seems sketch at best.